vowpalwabbit/array_parameters_dense.h

```diff
@@ -87,6 +89,32 @@ class dense_parameters
   inline const weight& operator[](size_t i) const { return _begin[i & _weight_mask]; }
   inline weight& operator[](size_t i) { return _begin[i & _weight_mask]; }
 
+  // Definitions for aggregated learning
+  std::unordered_map<uint64_t, std::bitset<32>> feature_bit_vector;
+  struct tag_hash_info
+  {
+    uint64_t tag_hash;
+    bool is_set=false;
+  };
+
+  void set_tag(uint64_t tag_hash)
```

Comment:
try writing a unit test [here](https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/test/unit_test/weights_test.cc) where you can manually create and set 10 tag hashes, after each one call `turn_on_bit` and assert that `is_activated` is true to make sure that this functionality is correct